src/bin/server.rs

```diff
@@ -51,36 +53,40 @@ extern crate yaml_rust;
 extern crate log;
 extern crate simple_logger;
 extern crate mio;
+extern crate mio_extras;
 extern crate hex;
 
 use std::env;
 use std::process;
 use std::fs::File;
-use std::io::Read;
-use std::io;
+use std::io::{Read, ErrorKind};
 use std::time::Duration;
 use std::net::SocketAddr;
 use std::sync::Arc;
-use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
+use std::thread;
 
 use mio::{Poll, Token, Ready, PollOpt, Events};
 use mio::net::UdpSocket;
-use mio::timer::Timer;
+use mio_extras::timer::Timer;
 
 use byteorder::{LittleEndian, WriteBytesExt};
 
 use roughenough::{RtMessage, Tag, Error};
-use roughenough::{VERSION, CERTIFICATE_CONTEXT, MIN_REQUEST_LENGTH, SIGNED_RESPONSE_CONTEXT, TREE_LEAF_TWEAK};
+use roughenough::{VERSION, CERTIFICATE_CONTEXT, MIN_REQUEST_LENGTH, SIGNED_RESPONSE_CONTEXT};
 use roughenough::sign::Signer;
+use roughenough::merkle::*;
 
-use ring::{digest, rand};
+use ring::rand;
 use ring::rand::SecureRandom;
 
 use yaml_rust::YamlLoader;
 
 const MESSAGE: Token = Token(0);
 const STATUS: Token = Token(1);
 
+pub static NUM_RESPONSES: AtomicUsize = AtomicUsize::new(0);
```

Comment:
`NUM_RESPONSES` is a global `pub static` but being used mutably. The SCREAMING_SNAKE_CASE implies it's a constant (like `MESSAGE` and `STATUS` right above).

Seems like this is `pub static` to facilitate sharing of it between `polling_loop` and the benchmark in `main`. Is there some other way to share the counter without it being global?